lib/src/key_value_storage.dart

```diff
@@ -16,4 +16,8 @@ abstract class KeyValueStorage {
   Future<Set<String>> getKeys();
 
   Future<void> clearAll();
+
+  Future<void> setDouble(String key, double value);
```

Comment:
If we have a `set`, we probably also need a `get`. I would also place the setters/getters in one place, and all of the other ones like `clearAll`, `readAll` etc. after the getters/setters